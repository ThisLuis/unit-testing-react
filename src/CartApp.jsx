
export const CartApp = () => {
  return (
    <h1>CartApp!!!</h1>
  );
}
