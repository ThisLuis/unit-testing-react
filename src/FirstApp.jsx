import PropTypes from 'prop-types';

export const FirstApp = ( {title, subTitle, name} ) => {
    return (
        <>
            {/* <code>{JSON.stringify(newMessage)}</code> */}
            <h1>{ title }</h1>
            <p>{ subTitle }</p>
            <p>{ name }</p>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired
}

// Las defaultProps se ejecutan primero que las propTypes
FirstApp.defaultProps = {
    name: 'Luis Nieves!!!',
    subTitle: 'Subtitulo por defecto',
    // title: 'Titulo principal',
}
