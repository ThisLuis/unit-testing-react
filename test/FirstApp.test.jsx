import { render } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas en <FirstApp />', () => {
    test('debe de hacer match con el snapshopt', () => {

        const title = 'Hola, soy un test';
        const { container } = render(<FirstApp title={ title} />);
        expect( container ).toMatchSnapshot();
    });

    test('debe de mostrar el titulo en un h1', () => {

        const title = 'Hola, soy un test';
        const { container, getByText } = render(<FirstApp title={ title} />);
        expect( getByText(title)).toBeTruthy();

        // Verificar que se encuentre dentro de un h1
        const h1 = container.querySelector('h1');
        expect( h1.innerHTML ).toContain(title);
    });
})