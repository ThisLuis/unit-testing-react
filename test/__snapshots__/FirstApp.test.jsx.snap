// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshopt 1`] = `
<div>
  <h1>
     
    Hola, soy un test
     
  </h1>
  <p>
    Subtitulo por defecto
  </p>
  <p>
    Luis Nieves!!!
  </p>
</div>
`;
